libp2p/wire.nim

```diff
@@ -8,65 +8,72 @@
 ## those terms.
 
 ## This module implements wire network connection procedures.
-import chronos
+import chronos, stew/endians2
 import multiaddress, multicodec
 
 when defined(windows):
   import winlean
 else:
   import posix
 
-proc initTAddress*(ma: MultiAddress): TransportAddress =
+const
+  TRANSPMA* = mapOr(
+    mapAnd(IP, mapEq("udp")),
+    mapAnd(IP, mapEq("tcp")),
+    mapAnd(mapEq("unix"))
+  )
+
+  RTRANSPMA* = mapOr(
+    mapAnd(IP, mapEq("tcp")),
+    mapAnd(mapEq("unix"))
+  )
+
+proc initTAddress*(ma: MultiAddress): MaResult[TransportAddress] =
```

Comment:
in most cases, it's good to attach `{.raises: [Defect].}` to things that return `Result` - is that possible here?

Reply:
Done